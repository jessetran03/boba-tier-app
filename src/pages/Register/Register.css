.register h2 {
  margin-bottom: 30px;
}

.register label {
  display: block;
  margin: 2px auto;
}

.register input {
  font-size: 20px;
  display: block;
  margin: 2px auto;
  padding: 4px 12px;
  border-radius: 15px;
  width: 250px;
}

