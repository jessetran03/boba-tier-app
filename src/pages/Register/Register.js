import React, { Component } from 'react'
import config from '../../config'
import PropTypes from 'prop-types'
import './Register.css'

export default class Register extends Component {
  state = {
    error: null
  }
  static propTypes = {
    location: PropTypes.object,
    history: PropTypes.object
  }

  handleRegisterSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/login'
    history.push(destination)
  }

  handleRegister = e => {
    e.preventDefault()
    const newUser = {
      first_name: e.target['first-name'].value,
      last_name: e.target['last-name'].value,
      user_name: e.target['username'].value,
      password: e.target['password'].value,
    }
    fetch(`${config.API_ENDPOINT}/users`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(newUser),
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      })
      .then(res =>
        this.handleRegisterSuccess()
      )
      .catch(res => {
        this.setState({ error: res.error })
      })
  };

  render() {
    const { error } = this.state
    return (
      <>
        <form onSubmit={this.handleRegister} className='register'>
          <h2>Sign up</h2>
          <div role='alert'>
            {error && <p className='form-error'>{error}</p>}
          </div>
          <label htmlFor='first-name-input'>
            First Name:
          </label>
          <input type='text' id='first-name-input' name='first-name' />
          <br />
          <label htmlFor='last-name-input'>
            Last Name:
          </label>
          <input type='text' id='last-name-input' name='last-name' />
          <br />
          <label htmlFor='username-input'>
            Username:
          </label>
          <input type='text' id='username-input' name='username' />
          <br />
          <label htmlFor='password-input'>
            Password:
          </label>
          <input type='password' id='password-input' name='password' />
          <br />
          <button type='submit'>
            Create account
          </button>

        </form>
      </>
    )
  }
}