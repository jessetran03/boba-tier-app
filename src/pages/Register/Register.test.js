import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Register from './Register'
import renderer from 'react-test-renderer'

it('renders Register without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <Register />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render Register without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <Register />
    </BrowserRouter>
  )
    .toJSON();
  expect(wrapper).toMatchSnapshot()
})