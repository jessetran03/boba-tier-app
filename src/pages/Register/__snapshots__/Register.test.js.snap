// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Register without data 1`] = `
<form
  className="register"
  onSubmit={[Function]}
>
  <h2>
    Sign up
  </h2>
  <div
    role="alert"
  />
  <label
    htmlFor="first-name-input"
  >
    First Name:
  </label>
  <input
    autoFocus={true}
    id="first-name-input"
    name="first-name"
    type="text"
  />
  <br />
  <label
    htmlFor="last-name-input"
  >
    Last Name:
  </label>
  <input
    id="last-name-input"
    name="last-name"
    type="text"
  />
  <br />
  <label
    htmlFor="username-input"
  >
    Username:
  </label>
  <input
    id="username-input"
    name="username"
    type="text"
  />
  <br />
  <label
    htmlFor="password-input"
  >
    Password:
  </label>
  <input
    id="password-input"
    name="password"
    type="password"
  />
  <br />
  <button
    type="submit"
  >
    Create account
  </button>
</form>
`;
